app/models/user/reputation_acquisition.rb

```diff
@@ -2,7 +2,19 @@ class User::ReputationAcquisition < ApplicationRecord
   belongs_to :user
   belongs_to :reason_object, polymorphic: true, optional: true
 
+  before_create do
+    raise ReputationAcquisitionReasonMissingAmount unless self.amount || REASON_AMOUNTS[self.reason]
+
+    self.amount = REASON_AMOUNTS[self.reason] unless self.amount
```

Comment:
I don't think amount should be overridable, so I think you can remove the `unless self.amount`